File: Chapter 10/JS/section_10_3_resilience_decorators/resilient/core/domain/orderStatus.js
/**
 * * @description
 * In Vanilla JS, we use a frozen object to simulate an Enum. 
 * This ensures that the system states are immutable and prevents 
 * "Magic String" contamination across the hexagon.
 */
export const OrderStatus = Object.freeze({
    PENDING_PAYMENT: "PENDING_PAYMENT",
    PAID: "PAID",
    FAILED: "FAILED"
});

File: Chapter 10/Python/section_10_3_resilience_decorators/resilient/core/domain/order_status.py
from enum import Enum

class OrderStatus(Enum):
    """
    DESIGN NOTE:
    Strictly enforcing system states via Enums prevents "Magic String" contamination. 
    By defining this inside 'core/domain', we ensure the entire system speaks the 
    same language, regardless of which external cloud provider we use.
    """
    PENDING_PAYMENT = "PENDING_PAYMENT"
    PAID = "PAID"
    FAILED = "FAILED"